### PR TITLE
WIP Support aioredis==1.0

### LINKED ISSUE #345: aiocache compatible with aioredis 1.0.0b2
I created pull request #344 which implements support for aioredis 0.3.3 and 1.0.0b2.

### PR DESCRIPTION
**Pull request by argaen**
Closes #345 

Reduces coverage but because its not being combined, this will be fixed in #361

### CODE REVIEW

File: aiocache/backends/redis.py [hunk 1]
```diff
@@ -65,12 +65,16 @@ def __init__(
...
+        if AIOREDIS_BEFORE_ONE:
+            return await self._pool.acquire()
+        return (await self._pool)
 
     async def release_conn(self, _conn):
-        self._pool.release(_conn)
+        if AIOREDIS_BEFORE_ONE:
+            self._pool.release(_conn)
+        else:
+            self._pool._pool_or_conn.release(_conn._pool_or_conn)
```
Comment (argaen) [inline aiocache/backends/redis.py]: meh.. need to check if there is a better way to release a connection...

File: aiocache/backends/redis.py [hunk 2]
```diff
@@ -185,7 +189,8 @@ def __init__(
 
     async def _close(self, *args, **kwargs):
         if self._pool is not None:
-            await self._pool.clear()
+            if AIOREDIS_BEFORE_ONE:
```
Comment (argaen) [inline aiocache/backends/redis.py]: There is no clear anymore? :thinking:

File: tests/acceptance/test_base.py [hunk 3]
```diff
@@ -267,4 +268,5 @@ class TestRedisCache:
     async def test_close(self, redis_cache):
         await redis_cache.set(pytest.KEY, "value")
         await redis_cache._close()
-        assert redis_cache._pool.size == 0
+        if AIOREDIS_BEFORE_ONE:
+            assert redis_cache._pool.size == 0
```
Comment (argaen) [inline tests/acceptance/test_base.py]: If clear is available somewhere, need to update here too so size is checked

File: tests/ut/backends/test_redis.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 from aiocache import RedisCache
 from aiocache.base import BaseCache
 from aiocache.serializers import JsonSerializer
-from aiocache.backends.redis import RedisBackend, conn
+from aiocache.backends.redis import RedisBackend, conn, AIOREDIS_BEFORE_ONE
 
 
 class FakeRedisConn:
```
Comment (argaen) [inline tests/ut/backends/test_redis.py]: Need to revisit this and `FakePool`. The whole thing is starting to become a mess and hard to understand. Maybe its worth to just separate in 2 classes. One for each version of redis and then let the fixture decide when to load one or the other

File: tests/ut/backends/test_redis.py [hunk 5]
```diff
@@ -345,7 +347,8 @@ def test_setup_override(self):
         cache._pool = pool
         await cache._raw("set", pytest.KEY, 1)
         await cache._close()
-        assert pool.clear.call_count == 1
+        if AIOREDIS_BEFORE_ONE:
```
Comment (argaen) [inline tests/ut/backends/test_redis.py]: same here for the clear thing

**Comment by argaen** [comment]
Hey @popravich do you mind having a look at this PR? Updated aiocache to support new aioredis release but there are couple of things that maybe I'm doing wrong (see the comments)

**Comment by popravich** [comment]
Hi @argaen ,
In your case, instead of using `create_redis_pool` and lots of `if AIOREDIS_BEFORE_ONE`
I would recommend using raw pool and `Redis` class so you will have less changes to your code,
here is the example of what I mean:
```python
# before v1.0
# self._pool = await create_redis(...)
redis = await self._pool.acquire()
self._pool.release(redis)

# since v1.0
# self._pool = await create_pool(...)
conn = await self._pool.acquire()
redis = Redis(conn)
self._pool.release(redis.connection)  # connection property returns _pool_or_conn attribute
```

**Comment by argaen** [comment]
@popravich done the changes according to your recommendation: https://github.com/argaen/aiocache/pull/360/files#diff-0509a8110645553d3299c7430fd7c554

Let me know if that look reasonable and I'll merge :)

**Comment by popravich** [comment]
Looks good

**Pull request merged**